### ISSUE TITLE
Support React 16

### ISSUE THREAD

**Issue by Graham42** [ReactTraining/react-idle#17]
Would like to have support for React 16. 
However do you still want to keep support for React 15? (I'm assuming yes) 
And if so, what's the strategy for testing both versions of React? (assuming that https://github.com/ReactTraining/react-idle/pull/10 will get merged)

I can do this work at some point, just wanted to open issue first.

**Comment by alacret** [comment]
It does not work. nom install on a 3.0.0 version using react 16.* won't install

**Comment by newyork-anthonyng** [comment]
@alacret What error are you seeing?

I created this [demo on Github](https://github.com/newyork-anthonyng/react-idle-test) using `create-react-app` (which uses `React v16`) and `react-idle v3`. Everything seems to work.

The only thing I'm seeing is a `warning` about a peer dependency to `React v15`, but everything still works.

**Comment by alacret** [comment]
Yes, we know that works, but the "peer" dependency stills makes a warning, and in older versions of nom it won't complete the install. My Jenkins and my buddy.works neither one can install my project. I manually change the "peer" rule and it work. But as this is not a solution we remove the library until a new version with the correction comes out
